### PR TITLE
Opentracing it

### PR DESCRIPTION
**Pull request by loicmathieu**
This PR creates a Smallrye OpenTracing integration test (inspired by the OpenTelemetry one) and include a specific test for the JDBC instrumentation.

This is still a work in progress as the JDBC span is not created, but the test crash if we use the current JDBC driver version and pass if we use version 2.0.4 so it might have catch #18033 if we had it previously.

### CODE REVIEW

**Comment by loicmathieu** [comment]
@kenfinnigan if you can have a look at it, it's still a draft as I need to find why the JDBC span is not created.

File: integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.common.mapper.TypeRef;
+
+@QuarkusTest
+public class OpenTracingTestCase {
+
+    static GenericContainer postgreSQLContainer;
+
+    @BeforeAll
```
Comment (kenfinnigan) [inline integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java:27]: Does it also need an `@AfterAll` to cleanup the container?

**Review by kenfinnigan — COMMENTED** [review]
Seems reasonable from a first look through

File: integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java (hunk 1 above)
Comment (loicmathieu) [reply integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java:27]: Testcontainers should stop it automatically (thanks to Ryuk container) but I can add one to force stopping the container eagerly

**Comment by kenfinnigan** [comment]
@loicmathieu Have you seen #18313? Are you ok with it?

**Comment by loicmathieu** [comment]
@kenfinnigan no, I'm OK to merge my changes on the new IT created for Zipkin compatibility when merged.
So let's just merge #18313 when you're OK with it, I'll rebase my changes on top of it.

**Comment by loicmathieu** [comment]
@kenfinnigan I rebase my PR onto main now that zipkin compatibility mode has been merged.
I choose to remove the existing IT that only test zipkin compatibility as my IT test way more functionality and both are not compatible (I use the mock support for easier testing).

If you think testing zipkin compatibility is important, I can copy the existing test and include it as a QuarkusUnitTest inside the deployment module of quarkus-jaeger.

File: integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+            if (spanData.get("operation_name").equals("GET:io.quarkus.it.opentracing.JdbcResource.jdbc")) {
+                Assertions.assertEquals(0, spanData.get("parent_spanId"));
+                Assertions.assertEquals("server", spanData.get("tag_span.kind"));
+                Assertions.assertEquals("jaxrs", spanData.get("tag_component"));
+                Assertions.assertEquals("GET", spanData.get("tag_http.method"));
+                Assertions.assertEquals("200", spanData.get("tag_http.status_code"));
+                resourceSpanFound = true;
+            }
+            //            else if (spanData.get("operation_name").equals("JDBC:io.quarkus.it.opentracing.JdbcResource.jdbc")) {
```
Comment (kenfinnigan) [inline integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java:308]: Is the JDBC span not working?

**Comment by kenfinnigan** [comment]
If you could add the Zipkin compatibility test back in the deployment I think that would be good. Thanks

File: integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java (hunk 2 above)
Comment (loicmathieu) [reply integration-tests/smallrye-opentracing/src/test/java/io/quarkus/it/opentracing/OpenTracingTestCase.java:308]: Well, it should :(
I'll have a look my old PR should be a WIP

**Comment by loicmathieu** [comment]
@kenfinnigan 

> If you could add the Zipkin compatibility test back in the deployment I think that would be good.

OK, will do it

**Comment by loicmathieu** [comment]
@kenfinnigan I fix the JDBC test.

For Zipkin, there is already a test inside the jaeger extension: https://github.com/quarkusio/quarkus/blob/main/extensions/jaeger/deployment/src/test/java/io/quarkus/jaeger/test/QuarkusJaegerTracerTest.java#L29

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**